source/math/FloatMatrix.ooc

```diff
@@ -141,30 +145,31 @@ FloatMatrix : cover {
 		result := 0.0f
 		for (i in 0 .. this height)
 			result += this[i, i]
+		this free(Owner Callee)
 		result
 	}
 
 	// <summary>
 	// Swaps the position of two rows
 	// </summary>
-	// <param name="row1">First row</param>
-	// <param name="row2">Second row</param>
-	swaprows: func@ (row1, row2: Int) {
-		version (safe) {
-			if (row1 < 0 || row2 < 0 || row1 >= this height || row2 >= this height)
-				raise("Invalid row choices in FloatMatrix swaprows")
-		}
+	// <param name="first">First row</param>
+	// <param name="second">Second row</param>
+	swapRows: func@ (first, second: Int) {
 		order := this order
 		buffer: Float
-		if (row1 != row2)
+		if (first != second)
 			for (i in 0 .. order) {
-				buffer = this elements[i + row1 * this width]
-				this elements[i + row1 * this width] = this elements[i + row2 * this width]
-				this elements[i + row2 * this width] = buffer
+				buffer = this elements[i + first * this width]
+				this elements[i + first * this width] = this elements[i + second * this width]
+				this elements[i + second * this width] = buffer
 			}
 	}
-
-	toString: func@ -> String {
+	// TODO: DEPRECATED
+	swaprows: func@ (first, second: Int) {
+		c"FloatMatrix swaprows deprecated" println()
+		this swapRows(first, second)
+	}
```

Comment:
`swapRows` is added in this PR not `swaprows` and since it is public I have to assume that somebody uses it.